doc/system-ready/system-ready-HLD.md

```diff
@@ -0,0 +1,482 @@
+    - Any service type other than oneshot if by default goes to inactive state, RemainAfterExit=yes entry needs to be added to their service unit file to be inline with the framework.
+
+
+## 4.5 Provision for docker apps to mark closest UP status
+
+The feature provides framework for docker services to mark its closest UP status. 
+
+### 4.5.1 CONFIG_DB Changes (init_cfg.json)
+
+Docker apps marking their UP status in STATE_DB will input an entry in FEATURE table of CONFIG_DB with check_up_status flag set to True through /etc/sonic/init_cfg.json file change.
+Sysmonitor checks for the check_up_status flag in CONFIG_DB before reading the app ready status from STATE_DB. 
+If the flag does not exist or if set to False, then sysmonitor will not read the app ready status but just checks the running status of the service.
+
+
+```
+- Schema in /etc/sonic/init_cfg.json
+  This json file will be fed to FEATURE table of CONFIG_DB during bootup.
+  This json file will be factory default and no config command will be provided for "check_up_status" entry to be updated in CONFIG_DB later.
+           {
+              "FEATURE": {
+                 "<dockername>": {
+                     ...
+                     "state": "enabled",
+                     "check_up_status": "True"
+                 }
+              }
+            }
+```
+
+### 4.5.2 STATE_DB Changes
+- Docker apps which rely on config, can mark UP_STATUS to True in STATE_DB  when they are ready to receive configs from CONFIG_DB and/or some extra dependencies are met.
+- Respective apps should mark its UP_STATUS considering Port ready status. Hence there is no separate logic check needed by system monitoring tool
+- Along with UP_STATUS, docker apps should update the FAIL_REASON with appropriate reason in case of failure or empty string in case of success.
+- Also, TIME to be fed in as well in the format of '%Y%m%d %H:%M:%S'.
+
+
+For instances,
+- swss docker app is considered to be UP only when its systemd service is in running state and when its dependencies are met by checking state-db entries populated by *mgrs, like intfmgr, vrfmgr, vxlanmgr.
+- Other apps like udld,stp etc once after waiting for PortInitDone status and interface db creation, they could mark their UP status.
+- Database app which is the first/base app to be up, may set the UP_STATUS to True once all the required number of redis-server instances are in running state. 
+  However, as the supervisor monitors all the redis db instances as part of critical process list, there is no need of a separate app ready status in database docker.
+
+
+STATE_DB:
+```
+- sonic-db-cli STATE_DB HSET "FEATURE|<dockername>" UP_STATUS True
+- sonic-db-cli STATE_DB HSET "FEATURE|<dockername>" FAIL_REASON "<some reason in string format>" / ""
+- sonic-db-cli STATE_DB HSET "FEATURE|<dockername>" TIME "<timestamp in <datetime.now().strftime('%Y%m%d %H:%M:%S')> format >"
+
+- Schema in STATE_DB
+  sonic-db-dump -n STATE_DB output
```

Comment:
The host services don't belong to FEATURE table and we need a different table for host services like hostcfgd

Reply:
Agreed.